include/obstack.h

```diff
@@ -135,11 +132,50 @@
  *   h: pointer to the handle used to grow the object.
  *   data: byte to be added
  *
- * Returned Value:
- *   Added byte.
+ ****************************************************************************/
+#define obstack_1grow_fast(h, data) ((void)(*((h)->next_free++) = (data)))
+
+/****************************************************************************
+ * Name: obstack_ptr_grow_fast
+ *
+ * Description:
+ *   Adds one pointer to the currently growing object.
+ *   There is no check if there is enough room and thus it is easy to cause
+ *   buffer overrun. Use only when you are sure that there is enough room!
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *   ptr: pointer to be added
+ *
+ ****************************************************************************/
+#define obstack_ptr_grow_fast(h, ptr) \
+  ({ \
```

Comment:
change to `do while`, since `({})` is a gcc special extension.

Reply:
Ok, but there are already palces where `({})` are used such as `nuttx/list.h` and `nuttx/spinlock.h` (although those can be disabled). I have nothing against changing that only that not using that extension might be hard either way.

Reply:
it's better to avoid the compiler special extension as much as we can.